components/scream/scripts/test_all_scream.py

```diff
@@ -301,10 +336,11 @@ def run_all_tests(self):
 
         for t,s in tests_success.items():
             if not s:
-                name = self._test_full_names[t]
-                print("Build type {} failed. Here's a list of failed tests:".format(name))
-                out = run_cmd("cat ctest-build/{}/Testing/Temporary/LastTestsFailed*".format(name))[1]
-                print(out)
+                print("Build type {} failed. Here's a list of failed tests:".format(self._test_full_names[t]))
+                test_dir = self.get_test_dir(t)
+                test_results_dir = pathlib.Path(test_dir, "Testing", "Temporary")
+                for result in test_results_dir.glob("LastTestsFailed*"):
```

Comment:
Nice.  We needed something like this. This somewhat addresses #219, although not completely. Do you think it would be too hard to re-run failed tests in case they're no more than, say, 2 per build? I don't want to re-run everything in case all tests failed, but if it's only 1 or 2, it may help to re-run them (`ctest --rerun-failed`). We don't need parallelization for this, we can simply do it serially, given that it's 6 (2x3) tests tops.

Reply:
Hey luca, this code is semantically identical to what was there before; it just uses python instead of the shell command. I like the idea of a re-run feature; we should add that to the list of features we want.